test/pool.test.ts

```diff
@@ -23,379 +23,371 @@ import {
 
 const CYCLE_BLOCKS = 10;
 
+// The next transaction will be executed on the first block of the next cycle,
+// but the next call will be executed on the last block of the current cycle
+async function mineBlocksUntilCycleEnd(): Promise<void> {
+  const blockNumber = await buidler.ethers.provider.getBlockNumber();
+  await mineBlocks(CYCLE_BLOCKS - ((blockNumber + 1) % CYCLE_BLOCKS));
+}
+
 type AnyPool = EthPool | Erc20Pool;
 
+abstract class PoolUser {
+  pool: AnyPool;
+  addr: string;
```

Comment:
A comment to clarify that this is the address of the user (and not the pool) would be helpful.

Reply:
Fixed